src/screens/Customers/AddMeterReading.tsx

```diff
@@ -1,61 +1,101 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { RouteComponentProps, Redirect, useHistory } from 'react-router-dom';
 import BaseScreen from '../../components/BaseComponents/BaseScreen';
 import Button from '../../components/Button';
 import TextField from '../../components/TextField';
+import { CustomerRecord } from '../../lib/airtable/interface';
+import { selectCurrentCustomer } from '../../lib/redux/customerData';
+import { EMPTY_METER_READING } from '../../lib/redux/customerDataSlice';
+import moment from 'moment';
+import { selectCurrentUserId } from '../../lib/redux/userData';
+import { getTariffPlanByCustomer, calculateAmountBilled } from '../../lib/utils/customerUtils';
+import { createMeterReadingAndUpdateCustomerBalance } from '../../lib/airtable/request';
+import OutlinedCardList from '../../components/OutlinedCardList';
+import { formatDateStringToLocal } from '../../lib/moment/momentUtils';
+
 
 const styles = (theme: Theme) =>
   createStyles({
-    content: {
-      color: theme.palette.text.primary,
-    },
-    outlined: {
-      padding: '15px 15px 10px',
-      border: `1px solid ${theme.palette.text.secondary}`,
-      borderRadius: '6px',
-      marginTop: '10px',
-    },
-    secondaryText: {
-      color: theme.palette.text.secondary,
+    amountOwedContainer: {
+      margin: '20px 0px'
     },
   });
 
 interface AddMeterReadingProps extends RouteComponentProps {
-  classes: { content: string; outlined: string; secondaryText: string };
-  location: any;
+  classes: { amountOwedContainer: string; };
 }
 
 function AddMeterReading(props: AddMeterReadingProps) {
   const { classes } = props;
+  const history = useHistory();
+  const currentCustomer: CustomerRecord | undefined = useSelector(selectCurrentCustomer);
+  const userId = useSelector(selectCurrentUserId);
+  const [meterReadingAmount, setMeterReadingAmount] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  if (!currentCustomer || !userId) {
+    return <Redirect to={'/customers'} />
+  }
+
+  const tariffPlan = getTariffPlanByCustomer(currentCustomer);
+
+  if (!tariffPlan) {
+    console.error("(Meter Readings) Could not find customer tariff plan, redirecting to Customer Main")
+    return <Redirect to={'/customers'} />
+  }
+
+  const startingMeterAmount = currentCustomer.startingMeterReading;
+  const startingMeterLastRecorded = currentCustomer.startingMeterLastChanged;
+
+
+  const handleSetMeterReadingAmount = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setMeterReadingAmount(event.target.value as string);
+  }
+
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    const currentReadingAmount = parseFloat(meterReadingAmount);
+    if (isNaN(currentReadingAmount)) {
+      setErrorMessage('Meter Reading Amount must be a number');
```

Comment:
Not too sure why this is the error message. Since TextField limits it to numeric inputs, the only things I could get to trigger this error were like '-.' or  '--'.